alpenhorn/acquisition.py

```diff
@@ -63,6 +178,99 @@ class FileType(base_model):
     name = pw.CharField(max_length=64)
     notes = pw.TextField(null=True)
 
+    # This dict is an atribute on the class used to hold the set of registered
+    # handlers. Store as a dictionary for easy lookup of handlers by name.
+    _registered_file_types = {}
+
+    @classmethod
+    def register_type(cls, file_info):
+        """Register a new file type with alpenhorn.
+
+        This creates the entry in the FileType table if it does not already exist.
+
+        Parameters
+        ----------
+        file_info : AcqInfoBase
+            AcqInfoBase describing the type of acquisition.
+        """
+
+        try:
+            cls.get(name=file_info._file_type)
+        except pw.DoesNotExist:
+            log.info("Create AcqType entry for \"%s\"" % file_info._file_type)
+            cls.create(name=file_info._file_type)
+
+        # Add to registry
+        cls._registered_file_types[file_info._file_type] = file_info
+
+    @classmethod
+    def check_registration(cls):
+        """Check that all FileTypes known to the database have a
+        registered handler.
+        """
+
+        # Get the list of types names from the database
+        db_file_types = [row[0] for row in cls.select(cls.name).tuples()]
+
+        # Get the names of all the registered types
+        reg_file_types = cls._registered_file_types.keys()
+
+        # Find any missing types
+        missing_file_types = set(db_file_types) - set(reg_file_types)
+
+        if len(missing_file_types):
+            raise RuntimeError('FileTypes %s have no registered handler.' %
+                               repr(missing_file_types))
+
+    def is_type(self, filename, acq, node):
+        """Check if this file can be handled by this file type.
+
+        Parameters
+        ----------
+        filename : string
+            Name of the file.
+        acq : ArchiveAcq
+            The acquisition the file is in.
+        node : StorageNode
+            The node we are importing from. Needed so we can inspect the actual
+            acquisition.
+
+        Returns
+        -------
+        is_type : boolean
+        """
+        return self.file_info._is_type(filename, path.join(node.root, acq.name))
```

Comment:
I'm still confused why acq- and file-type handlers need to know about the node root in relation to the acquisition name or file path? Isn't it potentially different on every node? It seems to me like the all of the paths should be root-relative, and then the code that operates on actual physical files on the disk simply prefixes it with the root of the node where it's operating.

I know this specific methods usually run only on a single node that does the importing (e.g., at DRAO), but don't we want to allow in principle multiple import locations, for instance if some expensive post-processing or analysis is done elsewhere?